src/types.ts

```diff
@@ -0,0 +1,258 @@
+   *
+   * To disable video autoplay, the autoplay attribute needs to be removed altogether as this attribute. Setting autoplay="false" will not work; the video will autoplay if the attribute is there in the <stream> tag.
+   *
+   * In addition, some browsers now prevent videos with audio from playing automatically. You may add the mute attribute to allow your videos to autoplay. For more information, [go here](https://webkit.org/blog/6784/new-video-policies-for-ios/).
+   */
+  autoplay: boolean;
+  /**
+   * Shows the default video controls such as buttons for play/pause, volume controls. You may choose to build buttons and controls that work with the player.
+   */
+  controls: boolean;
+  /**
+   * Returns the current playback time in seconds. Setting this value seeks the video to a new time.
+   */
+  currentTime: number;
+  /**
+   * A Boolean attribute; if included the player will automatically seek back to the start upon reaching the end of the video.
+   */
+  loop: boolean;
+  /**
+   * A Boolean attribute which indicates the default setting of the audio contained in the video. If set, the audio will be initially silenced.
+   */
+  muted: boolean;
+  /**
+   * Pause the video
+   */
+  pause: () => void;
+  /**
+   * Attempts to play the video. Returns a promise that will resolve if playback begins successfully, and rejects if it fails. The most common reason for this to fail is browser policies which prevent unmuted playback that is not initiated by the user.
+   */
+  play: () => Promise<void>;
+  /**
+   * A URL for an image to be shown before the video is started or while the video is downloading. If this attribute isn’t specified, a thumbnail image of the video is shown.
+   */
+  poster?: string;
+  /**
+   * This enumerated attribute is intended to provide a hint to the browser about what the author thinks will lead to the best user experience. You may choose to include this attribute as a boolean attribute without a value, or you may specify the value preload="auto" to preload the beginning of the video. Not including the attribute or using preload="metadata" will just load the metadata needed to start video playback when requested.
+   *
+   * The <video> element does not force the browser to follow the value of this attribute; it is a mere hint. Even though the preload="none" option is a valid HTML5 attribute, Stream player will always load some metadata to initialize the player. The amount of data loaded in this case is negligable.
+   */
+  preload: Preload;
+  /**
+   * Unsubscribe from events
+   */
+  removeEventListener: (event: string, handler: () => void) => void;
+  /**
+   * Either the video id or the signed url for the video you’ve uploaded to Cloudflare Stream should be included here.
+   */
+  src: string;
+  /**
+   * Number representing the intrinsic height of the video in pixels.
```

Comment:
Might be worth lingering to describe what `intrinsic height of the video` means, so its obvious to folks unfamiliar with the distinction.